### PR TITLE
Add test coverage to handler and return sequence classes

### PR DESCRIPTION
**Pull request by unfulvio-godaddy**
# Summary 

Adds test coverage for the `Handler` and `ReturnSequence` classes. 
These classes were refactored and moved into the `WP_Mock/Functions` namespace. 

Even though these methods are internal, it's a minor API break. However, we are considering to tag v1.0.0 soon so it should be acceptable.

The only change is for devs implementing their own function mocks outside of `WP_Mock::userFunction()` etc. by directly calling `\WP_Mock\Functions\Handler::handleFunction(__FUNCTION__, func_get_args());` -- this is documented in the docs.

## Details 

~~Note: this depends on #216 to be merged first.~~

## Contributor checklist 

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly 
- [x] I have added tests to cover changes introduced by this pull request
- [x] All new and existing tests pass

### Reviewer checklist 

 

- [x] Review and merge #216 first
- [ ] Code changes review
- [ ] Documentation changes review
- [ ] Unit tests pass
- [ ] Static analysis passes

### CODE REVIEW

File: php/WP_Mock/Functions.php [hunk 1]
```diff
@@ -105,12 +107,13 @@ public function register(string $function, array $args = [])
...
 
-        /** @var string $method */
+        /** @var callable-string $method */
         $method = preg_replace('/\\\\+/', '_', $function);
 
         $expectation = $this->setUpMock($mock, $method, $args);
 
-        Handler::register_handler($function, [$mock, $method]);
+        /** @phpstan-ignore-next-line */
+        Handler::registerHandler($function, [$mock, $method]);
```
Comment (nmolham-godaddy) [inline php/WP_Mock/Functions.php]: Is there a way to handle this phpstan without the ignore annotation?

File: php/WP_Mock/Functions/Handler.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+     * Handles a mocked function call.
+     *
+     * @param string|callable-string $functionName function name
+     * @param array<mixed> $args function arguments
+     * @return mixed
+     * @throws ExpectationFailedException
+     */
+    public static function handleFunction(string $functionName, array $args = [])
+    {
+        if (self::handlerExists($functionName)) {
```
Comment (nmolham-godaddy) [inline php/WP_Mock/Functions/Handler.php:48]: what do you think about using `stattic::` instead of `self::`?

Comment (unfulvio-godaddy) [reply php/WP_Mock/Functions/Handler.php:48]: I actually just moved this class from its original into a namespace and did not alter its contents

File: php/WP_Mock/Functions/Handler.php [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+        try {
+            self::handleFunction($functionName, $args);
+        } catch (Exception $exception) {
+            ob_end_clean();
+
+            throw $exception;
+        }
+
+        $result = ob_get_clean() ?: '';
```
Comment (nmolham-godaddy) [inline php/WP_Mock/Functions/Handler.php]: Do you expect the buttfer to be closed at anytime while handling the function? I see the catched expception get thrown right after ending/closing the buffer

Comment (unfulvio-godaddy) [reply php/WP_Mock/Functions/Handler.php]: the exception is caught but then thrown again before closing the buffer output

when not thrown, the buffer is captured

Comment (nmolham-godaddy) [reply php/WP_Mock/Functions/Handler.php]: So we don't need the `?: ''` in this case right?

```suggestion
        $result = ob_get_clean();
```

Comment (unfulvio-godaddy) [reply php/WP_Mock/Functions/Handler.php]: https://github.com/10up/wp_mock/pull/219/commits/71c133d45997f4522f048881c29fc5f7d30f2082 we can, and maybe it's a good idea to throw an exception if `ob_get_clean` returns false (not a string)

**Comment by unfulvio-godaddy** [comment]
note for reviewers:

as for this moment there are some GitHub issues also reported in https://www.githubstatus.com/ where some PR functions in GH are degraded - I'm pushing commits to this PR and they appear to hold in git (via CLI) but they don't show up in this PR and in the GH workflow - you may be able however to `git pull` the changes I have applied

File: php/WP_Mock/Functions.php (hunk 1 above)
Comment (unfulvio-godaddy) [reply php/WP_Mock/Functions.php]: yeah actually there is, I have updated the PR

**Review by nmolham-godaddy — APPROVED** [review]

**Pull request merged**